seacrowd/sea_datasets/udhr/udhr.py

```diff
@@ -0,0 +1,148 @@
+        schema=schema,
+        subset_id="udhr_{src}".format(src=src_lang),
+    )
+
+
+class NewDataset(datasets.GeneratorBasedBuilder):
+    """
+    The Universal Declaration of Human Rights (UDHR) is a milestone document in the history of
+    human rights. Drafted by representatives with different legal and cultural backgrounds from
+    all regions of the world, it set out, for the first time, fundamental human rights to be
+    universally protected. The Declaration was adopted by the UN General Assembly in Paris on
+    10 December 1948 during its 183rd plenary meeting.
+    """
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [seacrowd_config_constructor(lang, "source", _SOURCE_VERSION) for lang in _LANGS] + [seacrowd_config_constructor(lang, "seacrowd_ssp", _SEACROWD_VERSION) for lang in _LANGS]
+
+    DEFAULT_CONFIG_NAME = "udhr_ind_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_ssp":
+            features = schemas.self_supervised_pretraining.features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        urls = _URLS
+        data_dir = dl_manager.download_and_extract(urls)
+        lang = self.config.subset_id.split("_")[1]
+        if lang == "zlm":
+            lang = "mly_latn"
+
+        if lang == "cfm":
+            lang = "flm"
+
+        if lang == "hnj":
+            lang = "blu"
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, f"udhr_{lang}.txt".format(lang=lang)),
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        data = []
+        lang = self.config.subset_id.split("_")[1]
+
+        with open(filepath, "r") as f:
+            data = [line.rstrip() for line in f.readlines()]
+
+        if self.config.schema == "source":
+            yield 0, {"id": Path(filepath).stem, "text": " ".join(data)}
+
+        elif self.config.schema == "seacrowd_ssp":
+            yield 0, {"id": Path(filepath).stem, "text": " ".join(data)}
```

Comment:
I'm currently seeking some guidance from other owners on how this dataset should be loaded, loading one row per language does not feel right but will update here once I have the answer.

Reply:
Ok I got the answer, so rather than making this a different dataset, how about making each language a subset?

Reply:
> Ok I got the answer, so rather than making this a different dataset, how about making each language a subset?

Isn't this what has been done?

Reply:
I think what they want is that there is a single dataset containing all the language but with subset ids per language. Currently what has been done is each language becomes one dataset.

Reply:
Just checked some examples from huggingface, seems like this is the way to do subsets. Ignore my comment